zipline/data/us_equity_pricing.py

```diff
@@ -119,54 +104,112 @@ class NoDataOnDate(Exception):
     pass
 
 
-class BcolzDailyBarWriter(with_metaclass(ABCMeta)):
+def check_uint32_safe(value, colname):
+    if value >= UINT32_MAX:
+        raise ValueError(
+            "Value %s from column '%s' is too large" % (value, colname)
+        )
+
+
+def winsorise_uint32(df, column, *columns):
+    """Drops any record where a value would not fit into a uint32.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe to winsorise.
+    *columns : iterable[str]
+        The names of the columns to check.
+
+    Returns
+    -------
+    truncated : pd.DataFrame
+        ``df`` with rows that contain values that do not fit into a uint32
+        zeroed out.
+    """
+    columns = (column,) + columns
+    df[reduce(op.or_, (df[c] > UINT32_MAX for c in columns))] = 0
+    return df
+
+
+def ctable_from_dict(d, **bcolz_kwargs):
+    """
+    Construct a ctable from a dict mapping column name to numpy array.
+    """
+    names, columns = unzip(iteritems(d))
+    return ctable(columns=columns, names=names, **bcolz_kwargs)
+
+
+def to_ctable(raw_data):
+    if isinstance(raw_data, ctable):
+        # we already have a ctable so do nothing
+        return raw_data
+
+    columns = {}
+    for colname in OHLC:
+        coldata = raw_data[colname]
+        check_uint32_safe(coldata.max() * 1000, colname)
+        columns[colname] = coldata * 1000
+
+    winsorise_uint32(raw_data, 'volume', *OHLC)
```

Comment:
Do we want this to be the unconditional behavior?  I could imagine wanting to treat this as an error.  At the very least, we should probably fire a warning about dropped data.